lmdeploy/vl/model/llava.py

```diff
@@ -2,17 +2,21 @@
 # Modified from https://github.com/haotian-liu/LLaVA.git
 import ast
 import math
+import re
 import warnings
 from contextlib import contextmanager
 from typing import Dict, List
 
 import torch
 from PIL import Image
+from torch import nn
 from transformers import AutoConfig, AutoModelForCausalLM
 
 from lmdeploy.utils import get_logger
 from lmdeploy.vl.model.llava_hf import VISION_MODELS, LlavaHfVisionModel
-from lmdeploy.vl.model.utils import disable_logging, rewrite_ctx
+from lmdeploy.vl.model.utils import (disable_logging,
+                                     get_vision_encoder_device_map,
```

Comment:
`get_vision_encoder_device_map` is shared by `LlavaHfVisionModel`, `LlavaVisionModel` and `LlavaNextVisionModel`
I think it's better to put this function in the base class `LlavaHfVisionModel`